Fix `!6` host preference incorrectly allows IPv4 connections

taimoorzaeem opened the pull request:
The `HostIPv6Only` host preference was allowing IPv4 connections. This commits fixes this by setting a IPv6Only socket option which was required for it to work for IPv6 only connections.

I have tested this locally and this is working as expected. Related #78.

taimoorzaeem commented:
Hi @snoyberg!

Would it be possible for you to review this when you have time? I believe this shouldn't take much time to review because it's a pretty small change. Thanks in advance!

I can bump the version and add changelog as well, just let me know if that's okay.

snoyberg commented:
Hi @taimoorzaeem, for the most part I'm not actively maintaining my Haskell projects and I'm looking for comaintainers to step in for regular maintenance. I think this change should be fine though. Yes, a version bump and changelog entry addition would be great, and then I can merge and release.

taimoorzaeem commented:
> I think this change should be fine though. Yes, a version bump and changelog entry addition would be great, and then I can merge and release.

Done. Thank you in advance.

> Hi @taimoorzaeem, for the most part I'm not actively maintaining my Haskell projects and I'm looking for comaintainers to step in for regular maintenance.

@snoyberg I work on the https://github.com/PostgREST/postgrest project and I would be more than happy to be a co-maintainer, especially for the packages that are dependencies of `PostgREST`, as they often require upstream maintenance.

snoyberg commented:
Awesome, sounds great! I've added you as a maintainer on this repo. If you give me your Hackage username, I can add you there. And if there are other packages you'd be interested in maintaining, just let me know. I very much appreciate it :)

taimoorzaeem commented:
> Awesome, sounds great! I've added you as a maintainer on this repo. If you give me your Hackage username, I can add you there. And if there are other packages you'd be interested in maintaining, just let me know. I very much appreciate it :)

Many thanks @snoyberg! My hackage username is `taimoorzaeem`.

snoyberg commented:
Great, added!

<img width="1548" height="278" alt="image" src="https://github.com/user-attachments/assets/9dbf2d04-3a71-4207-bd76-25f3c1f6d6ae" />

taimoorzaeem committed:
Fix `!6` host preference incorrectly allows IPv4 connections

The `HostIPv6Only` host preference was allowing IPv4 connections.
This commits fixes this by setting a IPv6Only socket option which
was required for it to work for IPv6 only connections.

Signed-off-by: Taimoor Zaeem <taimoorzaeem@gmail.com>
